web/src/utils/network.ts

```diff
@@ -0,0 +1,209 @@
+  return stateKey.toLowerCase();
+};
+
+/**
+ * Check if an IP is valid
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An IP Address
+ * @return true if given IP is valid; false otherwise.
+ */
+const isValidIp = (value: IPAddress["address"]): boolean =>
+  ipaddr.IPv4.isValidFourPartDecimal(value);
+
+/**
+ * Check if a value is a valid netmask or network prefix
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An netmask or a network prefix
+ * @return true if given IP is valid; false otherwise.
+ */
+const isValidIpPrefix = (value: IPAddress["prefix"]): boolean => {
+  const prefix = value as string;
+
+  if (prefix.match(/^\d+$/)) {
+    return parseInt(prefix) <= 32;
+  } else {
+    return ipaddr.IPv4.isValidFourPartDecimal(prefix);
+  }
+};
+
+/**
+ * Prefix for the given prefix or netmask
+ *
+ * @note By now, only IPv4 is supported.
+ *
+ * @param value - An netmask or a network prefix
+ * @return prefix for the given netmask or prefix
+ */
+const ipPrefixFor = (value: string): number => {
+  if (value.match(/^\d+$/)) {
+    return parseInt(value);
+  } else {
+    return ipaddr.IPv4.parse(value).prefixLengthFromSubnetMask();
+  }
+};
+
+/**
+ *  Converts an IP given in decimal format to text format
+ *
+ * FIXME: IPv6 is not supported yet.
+ *
+ * @param address - An IP Address as network byte-order
+ * @return given address as string, when possible
+ */
+const intToIPString = (address: number): string | null => {
+  const ip = ipaddr.parse(address.toString());
+  if ("octets" in ip) {
+    return ip.octets.reverse().join(".");
+  } else {
+    return null;
+  }
+};
+
+/** Convert a IP address from text to network byte-order
+ *
+ * FIXME: Currently it is assumed 'le' ordering which should be read from NetworkManager State
+ *
+ * @param text - string representing an IPv4 address
+ * @return IP address as network byte-order
+ */
+const stringToIPInt = (text: string): number => {
+  if (text === "") return 0;
+
+  const parts = text.split(".");
+  const bytes = parts.map((s) => parseInt(s.trim()));
+
+  let num = 0;
+  const shift = (b) => 0x100 * num + b;
+  for (const n of bytes.reverse()) {
+    num = shift(n);
+  }
+
+  return num;
+};
+
+/**
+ * Returns given IP address in the X.X.X.X/YY format
+ */
+const formatIp = (addr: IPAddress): string => {
+  if (addr.prefix === undefined) {
+    return `${addr.address}`;
+  } else {
+    return `${addr.address}/${addr.prefix}`;
+  }
+};
+
+const buildAddress = (address: string): IPAddress => {
+  const [ip, netmask] = address.split("/");
+  const result: IPAddress = { address: ip };
+  if (netmask) result.prefix = ipPrefixFor(netmask);
+  return result;
+};
+
+
+/**
+ * @param flags - AP flags
+ * @param wpa_flags - AP WPA1 flags
+ * @param rsn_flags - AP WPA2 flags
+ * @return supported security protocols
+ */
+const securityFromFlags = (flags: number, wpa_flags: number, rsn_flags: number): SecurityProtocols[] => {
```

Comment:
@teclator If I remember correctly, you told me that everything should be camelCase. Should we adapt this?